PCbuild/find_msbuild.bat

```diff
@@ -0,0 +1,49 @@
+@rem
+@rem Searches for MSBuild.exe. This is the only tool we need to initiate
+@rem a build, so we no longer search for the full VC toolset.
+@rem
+@rem This file is supposed to modify the state of the caller (specifically
+@rem the MSBUILD variable), so we do not use setlocal or echo, and avoid
+@rem changing any other persistent state.
+@rem
+
+@rem No arguments provided means do full search
+@if '%1' EQU '' goto :begin_search
```

Comment:
The quoting rules are horrible here, because I want to handle where users have done:

```
set MSBUILD=C:\path to\MSBuild.exe
set MSBUILD="C:\path to\MSBuild.exe"
```

Which breaks in a whole lot of ridiculous ways. Using single quotes here at least handles the two sane cases without causing syntax errors.